cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
I also tried IgnoreSIGPIPE, and was disappointed as you are :(.

Regarding the ordering guarantee (before dcos-download), we already have that. `local-fs-pre.target` is the first target that systemd starts (https://www.freedesktop.org/software/systemd/man/bootup.html). dcos-download (and the other dcos services) are effectively started from cloud-final.service (when it runs /var/lib/cloud/instance/scripts/runcmd), which runs in `multi-user.target` (well after `local-fs-pre.target`). There would also be some ambiguity using the dcos-* services or targets as dependencies, since many of them don't exist until dcos-setup.service runs.

The critical section idea is interesting. Another idea that I would like to explore is sending the output somewhere other than the journal so it doesn't matter if/when journald restarts. Ideally this could be done with `StandardOutput` and a tty, but a straightforward option would be to simply set the script's output descriptors with `exec`.

I'm going to try this out, and restore some of the /var/log logic as we've discussed here. Let me know what you think in the meantime.